README.md

```diff
@@ -1,237 +1,109 @@
+1. Create a localized `package.json` file in each packages' `cjs` directory with the contents `{ type: 'commonjs' }`.
+   This effectively overwrites the package's own `package.json` which is set to `module`.
 
-**Returns** an instance of the [ByuJWT](#constructor)
+   Node Resolution
+   Algorithm: [See ESM_FILE_FORMAT](https://nodejs.org/dist/latest-v18.x/docs/api/esm.html#resolver-algorithm-specification)
 
-### Authenticate
+   ESM & CommonJS Module Tutorial: https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html
 
-Check the headers to see if the requester is authenticated.
+2. Use a tool such as [unbuild](https://github.com/unjs/unbuild) which outputs files with the `.cjs` and `.mjs`
+   extensions.
 
-`ByuJWT.prototype.authenticate ( headers )`
+   Example of Using `unbuild`: https://github.com/unjs/radix3/blob/main/package.json
 
-**Parameters**
+   MDN Docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules
 
-- *headers* - An `object` representing the header names and values. This method is looking specifically for two headers:
+This library uses the first method for two reasons:
 
-    1. `x-jwt-assertion` is a header that contains the JWT for the current client.
+1. Some tools may never support the `.mjs` extension
+2. `(package.json).type` is more deterministic since the resolution algorithm is built into node and bundlers.
 
-    2. `x-jwt-assertion-original` is a header that contains the JWT for the original requester. This value should be set
-       if a client is making an authenticated request on behalf of a different client.
-
-**Returns** a promise that, if authenticated, resolves to an object with some of these properties:
-
-- *current* - The current client's [decoded JWT](#decode-jwt).
-
-- *original* - The original client's [decoded JWT](#decode-jwt). This property may not be defined.
-
-- *originalJWT* - The JWT string provided by the original requester, or if that doesn't exist then of the current
-  client.
-
-- *claims* - A [decoded JWT's](#decode-jwt) primary claim, prioritized in this order:
-
-    1. Original resource owner
-    2. Current resource owner
-    3. Original client
-    4. Current client
-
-### Authenticate University API Middleware
-
-A middleware that will check if the request has authentication and will either add the property `verifiedJWTs` to the
-request or will respond to the request with a `401` or `500` response code.
-
-`ByuJWT.prototype.authenticateUAPIMiddleware`
-
-**Parameters**
-
-- *req* - The request object.
-
-- *res* - The response object.
-
-- *next* - The next function.
+## Testing
 
-**Returns** `undefined`
+The choice to use [Ava](https://avajs.dev) was made because it
 
-```js
-const express = require('express')
-const byuJwt = require('byu-jwt')()
+1. supports TypeScript and ESM out of the box
```

Comment:
Ava's uncommon at BYU OIT (compared to Jest), but ESM support is the killer feature for this use case.

Reply:
Unfortunately I've only found one other competitor in this area of the ecosystem - [uvu](https://github.com/lukeed/uvu) which I don't think supports typescript out of the box like ava does.